go/pkg/depgraph/builder_test.go

```diff
@@ -96,10 +96,30 @@ func TestBuilder_Build(t *testing.T) {
 	builder, err := NewBuilder(&PkgManager{Name: "golang"}, nil)
 	require.NoError(t, err)
 
+	node := builder.AddNode("dep@1.0.0", &PkgInfo{Name: "dep", Version: "1.0.0"})
```

Comment:
Should AddNode (optionally) accept NodeInfo? I think it does in the Javascript lib